nanomq_cli/zmq_gateway.c

```diff
@@ -384,17 +384,18 @@ gateway_start(int argc, char **argv)
 		fprintf(stderr, "Memory alloc error.\n");
 		exit(EXIT_FAILURE);
 	}
+	proxy_info *info = NULL;
 
 	gateway_conf_init(conf);
 	gateway_parse_opts(argc, argv, conf);
 	conf_gateway_parse_ver2(conf);
 	if (conf->http_server.enable) {
-		proxy_info info = {
-			.proxy_name  = "zmq",
-			.conf        = &config,
-			.conf_path   = config.path,
-			.http_server = &config->http_server,
-		};
+		info->proxy_name  = "zmq";
```

Comment:
It may be better to use functions or macros for this part of the code.

Reply:
ok